core/field_image.js

```diff
@@ -171,8 +187,8 @@ Blockly.FieldImage.prototype.updateWidth = function() {
  * If field click is called, and click handler defined,
  * call the handler.
  */
- Blockly.FieldImage.prototype.showEditor = function() {
-   if (this.clickHandler_){
-     this.clickHandler_(this);
-   }
- };
+Blockly.FieldImage.prototype.showEditor_ = function() {
```

Comment:
oooh. This missing underscore was in my notes of stuff to investigate about fields! But now I'm a bit confused because showEditor should get called properly and we're binding mouseDown.  Does the callback then happen twice?

Semi-related question: is mousedown the right event? or is click or up better?

Reply:
I beleive showEditor showing up is handled by gesture code, in this case you would just be passing that over to the predefined click handler attached to an image, allowing you to take an action on image click. 
As for the specific mouse event thats handled, thats all taken care of by the gesture logic that calls showEditor

Reply:
Normally field `init` binds `mouseDown`, yes.  But `fieldImage` `init` doesn't call the superclass's `init` function, so the binding doesn't happen.

Binding to `mouseDown` is correct because the gesture code will figure out if it's a click and call `showEditor`.  Notice that the mouse down handler that is bound is actually the one from Field, which will set the start field on the gesture.  

So `showEditor` isn't called on mouse down--it's called when the Gesture decides it's a field click.

Reply:
Ah, yes. I didn't read the bind carefully enough - I thought you were binding this.clickHandler_ as the callback. Makes sense.  I will add the binding mouse events to my list of things to consider in a better api for fields as well.